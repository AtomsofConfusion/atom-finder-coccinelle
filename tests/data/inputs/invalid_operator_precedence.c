#include <stdio.h>

int main() {
  int a = 1;
  int b = 2;
  int c = 3;

  a || b & c;

  a || b ^ c;

  a || b + c;

  a || b - c;

  a || b << c;

  a && b | c;

  a && b + c;

  a && b - c;

  a && b << c;

  a || b && c;

  (a || b) && c; // should not include this non-confusing

  return 0;
}
