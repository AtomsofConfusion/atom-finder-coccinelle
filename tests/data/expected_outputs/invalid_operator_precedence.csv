atom,file,line,offset,code
operator-precedence,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/invalid_operator_precedence.c,12,105,a||b+c
operator-precedence,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/invalid_operator_precedence.c,14,120,a||b-c
operator-precedence,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/invalid_operator_precedence.c,18,151,a&&b|c
operator-precedence,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/invalid_operator_precedence.c,20,166,a&&b+c
operator-precedence,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/invalid_operator_precedence.c,22,181,a&&b-c
operator-precedence,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/invalid_operator_precedence.c,24,196,a&&b<<c
operator-precedence,/home/veering/Projects/atom-finder-coccinelle/tests/data/inputs/invalid_operator_precedence.c,26,212,a||b&&c
